### PR TITLE
Edouard/GameEngineTest

### PR DESCRIPTION
**Pull request by edischmitz**
wrote Test for the game Engine class these Test handle tasks and Test all main functionalities of all "public" methods of the GameEngine class:
#40 #39 #49 #215 #13 #11 #10

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java [hunk 1]
```diff
@@ -74,6 +74,7 @@ public void addUserToWaitingRoom(User user){
                     String userIdentity = userInWaitingRoom.getSessionIdentity();
                     webSocketService.sendGameAssignmentMessageToWaitingRoom(userIdentity, createdGame.getPlayers(), createdGame.getGameId());
                 }
+                waitingRoom.resetUsers();
```
Comment (pemmenegger) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java:77]: does this not already happening in waitingRoom.getXNumberOfUsers() with the line: User newUserToAddToGamesession = userQueue.remove(0); ? maybe we should better decrease the usercount directly in here after this line, or even better make a getUserCount() which actually computes the usercount every time from the current queue

File: src/test/java/ch/uzh/ifi/hase/soprafs21/objects/GameTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+
+    @Test
+    public void setPlayerToReadyTest(){
+        game.setPlayerToReady(game.getPlayers().get(0).getPlayerName());
+        assertNotEquals(game.getCurrentRound(), null);
+    }
+
+    @Test
+    public void executeMoveTest(){
```
Comment (pemmenegger) [inline src/test/java/ch/uzh/ifi/hase/soprafs21/objects/GameTest.java:55]: not needed, we have moveTests

File: src/test/java/ch/uzh/ifi/hase/soprafs21/objects/GameTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public void userToPlayerTest(){
+        Player player1 = game.userToPlayer(user1);
+        assertEquals(user1.getUsername(), player1.getPlayerName());
+    }
+
+    @Test
+    public void updatePlayerColorTest(){
+        game.updatePlayerColor(game.getPlayers().get(0).getPlayerName(), Color.RED);
+        assertEquals(game.getPlayers().get(0).getColor(), Color.RED);
+        //game.updatePlayerColor(game.getPlayers().get(0).getPlayerName(),null);
```
Comment (pemmenegger) [inline src/test/java/ch/uzh/ifi/hase/soprafs21/objects/GameTest.java:38]: should this be commented?

**Comment by edischmitz** [comment]
Corrections were made code should be ready to be merged with the dev

**Review by pemmenegger — APPROVED** [review]

**Pull request merged**